backend/models.py

```diff
@@ -100,6 +100,8 @@ def serialize_leaderboard(self, rank):
         return {
             "id": self.id,
             "username": self.username,
+            "firstname": self.first_name,
```

Comment:
NIT: Could've kept consistent and used `first_name` rather than `firstname`, but now you already used it in the frontend so no worries if you dont want to change it.